qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/upgrades/IgnoredFieldRollingUpgradeIT.java

```diff
@@ -67,10 +73,34 @@ public void testIgnoredMetaFieldExistsQuery() throws IOException {
         }
     }
 
+    @SuppressWarnings("unchecked")
+    public void testIgnoredMetaFieldGetQuery() throws IOException {
+        assumeTrue(
+            "_ignored metadata field used stored fields up to version 8.14, then started using doc values fields from version 8.15",
+            getOldClusterIndexVersion().before(IndexVersions.DOC_VALUES_FOR_IGNORED_META_FIELD)
+        );
+        if (isOldCluster()) {
+            assertRestStatus(client().performRequest(createNewIndex(INDEX_NAME)), RestStatus.OK);
+            assertRestStatus(client().performRequest(indexDocument(INDEX_NAME, "1", "foofoo")), RestStatus.CREATED);
+            final Map<String, Object> doc = entityAsMap(get(INDEX_NAME, "1"));
+            assertThat(((List<?>) doc.get(IgnoredFieldMapper.NAME)).get(0), Matchers.equalTo("keyword"));
+        } else if (isUpgradedCluster()) {
+            assertRestStatus(client().performRequest(indexDocument(INDEX_NAME, "2", "foobar")), RestStatus.CREATED);
+            final Map<String, Object> doc = entityAsMap(get(INDEX_NAME, "2"));
+            // NOTE: from version 8.15 _ignored is not returned anymore by default when using the get api since
+            // `get` only allows fetching stored fields via `stored_fields` and _ignored is not stored anymore.
+            assertNull(doc.get(IgnoredFieldMapper.NAME));
+        }
```

Comment:
Can you expand on this test? The get call below does not ask for the `_ignored` field explicitly. Am I getting confused or get did not return `_ignored` by default? 

Also, if you handle the two versions in the test, why execute it only against a specific version? That looks like something odd is happening?

Reply:
Yes `GET` used to return the `_ignored` field up to version `8.4.3` while it is no longer the case from version `8.5.0`